src/app/clusters/time-synchronization-server/time-synchronization-cluster.cpp

```diff
@@ -381,33 +510,25 @@ void TimeSynchronizationServer::OnTimeSyncCompletionFn(TimeSourceEnum timeSource
         }
         return;
     }
-    mGranularity  = granularity;
-    Status status = TimeSource::Set(kRootEndpointId, timeSource);
-    if (!(status == Status::Success || status == Status::UnsupportedAttribute))
-    {
-        ChipLogError(Zcl, "Writing TimeSource failed.");
-    }
+    mGranularity = granularity;
+    mTimeSource  = timeSource;
```

Comment:
Do we maybe need `NotifyAttributeChanged` calls here since attributes changed?

Ideally we would only notify if values did actually change, so we need a compare and only set & notify if the compare says the values changed.